fe/fe-core/src/main/java/com/starrocks/planner/CrossJoinNode.java

```diff
@@ -87,19 +108,73 @@ protected String debugString() {
     @Override
     protected void toThrift(TPlanNode msg) {
         msg.node_type = TPlanNodeType.CROSS_JOIN_NODE;
+        msg.cross_join_node = new TCrossJoinNode();
+        if (!buildRuntimeFilters.isEmpty()) {
+            msg.cross_join_node.setBuild_runtime_filters(
+                    RuntimeFilterDescription.toThriftRuntimeFilterDescriptions(buildRuntimeFilters));
+        }
     }
 
     @Override
     protected String getNodeExplainString(String detailPrefix, TExplainLevel detailLevel) {
-        StringBuilder output = new StringBuilder().append(detailPrefix + "cross join:" + "\n");
+        StringBuilder output = new StringBuilder().append(detailPrefix).append("cross join:").append("\n");
         if (!conjuncts.isEmpty()) {
-            output.append(detailPrefix + "predicates: ").append(getExplainString(conjuncts) + "\n");
+            output.append(detailPrefix).append("predicates: ").append(getExplainString(conjuncts) + "\n");
         } else {
-            output.append(detailPrefix + "predicates is NULL.\n");
+            output.append(detailPrefix).append("predicates is NULL.\n");
+        }
+        return output.toString();
+    }
+
+    @Override
+    protected String getNodeVerboseExplain(String detailPrefix) {
+        String explain = super.getNodeVerboseExplain(detailPrefix);
+        StringBuilder output = new StringBuilder(explain);
+        if (!buildRuntimeFilters.isEmpty()) {
+            output.append(detailPrefix).append("build runtime filters:\n");
+            for (RuntimeFilterDescription rf : buildRuntimeFilters) {
+                output.append(detailPrefix).append("- ").append(rf.toExplainString(-1)).append("\n");
+            }
         }
         return output.toString();
     }
 
+    @Override
+    public void buildRuntimeFilters(IdGenerator<RuntimeFilterId> generator) {
```

Comment:
What's difference to `JoinNode::buildRuntimeFilters`

Reply:
1. for cross join we should build RF for conjuncts other than join eq conjuncts.
2. in cross join conjuncts. we couldn't swap left child and right child

Reply:
Could we just to add some code in `JoinNode::buildRuntimeFilters`
- check if join node is cross join node
- if cross join node, then no need to swa left/right child.

I don't like to see a lot of biolterplate code, if it can be simplified by adding some if-condition stmt in `JoinNode::buildRuntimeFilters`.

Reply:
I move it to RuntimeFilterHelper